libpromises/locks.c

```diff
@@ -586,6 +604,13 @@ static bool KillLockHolder(const char *lock)
 
     CloseLock(dbp);
 
+    if (!IsCfengineLockHolder(lock_data.pid)) {
+        Log(LOG_LEVEL_VERBOSE,
+            "Lock holder with PID %ju was replaced by a non CFEngine process, ignoring request to kill it",
```

Comment:
Would be very nice if we printed the name of this process, so that it is possible spot cases where a request to an actual CFEngine process is ignored. However, no requirement in my opinion

Reply:
It's a bit more complicated to get it into this log message, but I added a debug log message to `IsCfengineLockHolder()`.